pandas/tools/plotting.py

```diff
@@ -359,7 +360,16 @@ def scatter_matrix(frame, alpha=0.5, figsize=None, ax=None, grid=False,
     density_kwds = density_kwds or {}
 
     # workaround because `c='b'` is hardcoded in matplotlibs scatter method
-    kwds.setdefault('c', plt.rcParams['patch.facecolor'])
+    # only set default for `c` if `color` kwarg not supplied
+    # edgecolors default required otherwise `color` has a border
+    # GH 14855
+    kwds.setdefault('edgecolors', 'none')
+    if 'color' in kwds:
+        pass
+    if 'facecolor' in kwds:
+        kwds.setdefault('facecolor', plt.rcParams['patch.facecolor'])
+    if 'c' in kwds:
+        kwds.setdefault('c', plt.rcParams['patch.facecolor'])
```

Comment:
Good thinking, @tacaswell! I investigated it a bit further and it turns out we can resolve this issue and the same issue for scatter_plot by removing the old bug fix, removing the new changes and adding in the edgecolor='none' as default. With this, we still get the identical behavior with `c`, `color` or `facecolor` set and no error.